python/dgl/nn/pytorch/gt/spatial_encoder.py

```diff
@@ -140,107 +142,63 @@ def __init__(self, num_kernels, num_heads=1, max_node_type=1):
-            sub_coord = coord[sum_num_nodes : sum_num_nodes + num_nodes]
-            # shape: [n, n], n = num_nodes
-            euc_dist = th.cdist(sub_coord, sub_coord, p=2)
-            if node_type is None:
-                # shape: [1]
-                mul = self.mul.weight[0, 0]
-                bias = self.bias.weight[0, 0]
-            else:
-                sub_node_type = node_type[
-                    sum_num_nodes : sum_num_nodes + num_nodes
-                ]
-                mul_embedding = self.mul(sub_node_type)
-                bias_embedding = self.bias(sub_node_type)
-                # shape: [n, n]
-                mul = mul_embedding[:, 0].unsqueeze(-1).repeat(
-                    1, num_nodes
-                ) + mul_embedding[:, 1].unsqueeze(0).repeat(num_nodes, 1)
-                bias = bias_embedding[:, 0].unsqueeze(-1).repeat(
-                    1, num_nodes
-                ) + bias_embedding[:, 1].unsqueeze(0).repeat(num_nodes, 1)
-            # shape: [n, n, k], k = num_kernels
-            scaled_dist = (
-                (mul * euc_dist + bias)
-                .repeat(self.num_kernels, 1, 1)
-                .permute((1, 2, 0))
-            )
-            # shape: [k]
-            gaussian_mean = self.gaussian_means.weight.float().view(-1)
-            gaussian_var = (
-                self.gaussian_stds.weight.float().view(-1).abs() + 1e-2
-            )
-            # shape: [n, n, k]
-            gaussian_kernel = (
-                (
-                    -0.5
-                    * (
-                        th.div(
-                            scaled_dist - gaussian_mean, gaussian_var
-                        ).square()
-                    )
-                )
-                .exp()
-                .div(-math.sqrt(2 * math.pi) * gaussian_var)
-            )
-
-            encoding = self.linear_layer_1(gaussian_kernel)
-            encoding = F.gelu(encoding)
-            # [n, n, k] -> [n, n, a], a = num_heads
-            encoding = self.linear_layer_2(encoding)
-            spatial_encoding[i, :num_nodes, :num_nodes] = encoding
-            sum_num_nodes += num_nodes
-        return spatial_encoding
+        bsz, N = coord.shape[:2]
+        euc_dist = th.cdist(coord, coord, p=2.0)  # shape: [B, n, n]
+        if node_type is None:
+            node_type = th.zeros([bsz, N, N, 2], device=coord.device).long()
+        else:
+            src_node_type = node_type.unsqueeze(-1).repeat(1, 1, N)
+            tgt_node_type = node_type.unsqueeze(1).repeat(1, N, 1)
+            node_type = th.stack(
+                [src_node_type + 2, tgt_node_type + self.max_node_type + 3],
+                dim=-1,
+            )  # shape: [B, n, n, 2]
+
+        # scaled euclidean distance
+        mul = self.mul(node_type).sum(dim=-2)  # shape: [B, n, n, 1]
+        bias = self.bias(node_type).sum(dim=-2)  # shape: [B, n, n, 1]
+        euc_dist = mul * euc_dist.unsqueeze(-1) + bias  # shape: [B, n, n, 1]
```

Comment:
Consider renaming the variables to gamma and beta.

Reply:
sure, fixed